django/db/models/deletion.py

```diff
@@ -469,11 +474,25 @@ def delete(self):
                     deleted_counter[qs.model._meta.label] += count
 
             # update fields
-            for model, instances_for_fieldvalues in self.field_updates.items():
-                for (field, value), instances in instances_for_fieldvalues.items():
+            for (field, value), instances_list in self.field_updates.items():
+                updates = []
+                objs = []
+                for instances in instances_list:
+                    if (
+                        isinstance(instances, models.QuerySet)
+                        and instances._result_cache is None
+                    ):
+                        updates.append(instances)
+                    else:
+                        objs.extend(instances)
+                if updates:
+                    combined_updates = reduce(or_, updates)
+                    combined_updates.update(**{field.name: value})
+                if objs:
+                    model = objs[0].__class__
```

Comment:
This might not be the best place to ask this, but is there any reason why this is not `field.model` instead?

When using polymorphism, this line causes database consistency too break, as it only updates the table of the derived class which corresponds to the first object in `objs`. However if it were `field.model` instead, the base class' table would be updated and consistency would be honored.

Should I create a PR?

Reply:
Interesting, please create a [new ticket](https://code.djangoproject.com/newticket) in Trac with a sample project and follow our [bug reporting guidelines](https://docs.djangoproject.com/en/dev/internals/contributing/bugs-and-features/#reporting-bugs). All bugfixes require [an accepted ticket](https://code.djangoproject.com/newticket).